File: ts/src/test/Exchange/test.fetchMarketLeverageTiers.ts
import assert from 'assert';
import { Exchange } from "../../../ccxt";
import testLeverageTier from './base/test.leverageTier.js';
import testSharedMethods from './base/test.sharedMethods.js';

async function testFetchMarketLeverageTiers (exchange: Exchange, skippedProperties: object, symbol: string) {
    const method = 'fetchMarketLeverageTiers';
    const tiers = await exchange.fetchMarketLeverageTiers (symbol);
    testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, tiers, symbol);
    for (let j = 0; j < tiers.length; j++) {
        testLeverageTier (exchange, skippedProperties, method, tiers[j]);
    }
    return true;
}

export default testFetchMarketLeverageTiers;

File: julia/Ccxt/test/exchange/test.fetchMarketLeverageTiers.jl
using Test
using Ccxt
function testFetchMarketLeverageTiers(exchange, skippedProperties, symbol)

    method = "fetchMarketLeverageTiers";
    tiers = Base.fetch(fetchMarketLeverageTiers(exchange, symbol));
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, tiers, symbol);
    j = 0
    while functions.ccxtruthy(functions.ccxt_lt(j, length(tiers)))
        testLeverageTier(exchange, skippedProperties, method, get(tiers, j + 1, nothing));
        j += 1
    end
    return true
end
